### PR TITLE
feat(react): test vs react 18 and react 17

### PR DESCRIPTION
**Pull request by stemey**

### CODE REVIEW

**Review by NSteinhoff — COMMENTED** [review]
Hi Stefan,

looks good at first glance, but I'd like to do run it locally first to check the demos for errors or warnings. That might take a little longer. If @clebert gives his approval, then feel free to go ahead.

I'm not entirely sure how breaking this is and which combinations of React versions we want to support. Why do we still want to test the feature-hub against React 17? What's the migration path to migrate a system to React 18, when you can't synchronize the deployment of the integrator with the deployment of the feature-apps? Could be a nothing-burger, but I'd like to be sure.

**Comment by stemey** [comment]
Hi @NSteinhoff ,
in general the migration relies on a feature hub version that works with 17 and 18, which can be used by all teams. Once all teams are ready for 18, the main integrator will switch from 17 to 18 ideally without changing any other code.

File: packages/core/src/__tests__/create-feature-hub.test.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ describe('createFeatureHub()', () => {
           featureAppManager.getAsyncFeatureAppDefinition(
             'http://example.com/test.js',
           ),
-        ).toThrowError(new Error('No module loader provided.'));
+        ).toThrow(new Error('No module loader provided.'));
```
[the remark this answers is not in the document]
Comment (stemey) [reply packages/core/src/__tests__/create-feature-hub.test.ts:52]: I got lint errors or something.

File: packages/react17/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@feature-hub/react17",
+  "version": "3.0.0",
+  "private": true,
+  "description": "A library for building a Feature Hub integrator with React.",
```
[the remark this answers is not in the document]
Comment (stemey) [reply packages/react17/package.json:5]: yes

File: packages/demos/src/react-loading-and-error-ui/integrator.tsx [hunk 3]
```diff
@@ -13,11 +13,11 @@ const {featureAppManager} = createFeatureHub('test:integrator', {
   providedExternals: {react: process.env.REACT_VERSION as string},
 });
 
-function ErrorUi({error}: {error: unknown}): JSX.Element {
+function ErrorUi({error}: {error?: unknown}): JSX.Element {
   return (
     <Callout intent={Intent.DANGER}>
       <H4>Example Error UI</H4>
-      <p>{error instanceof Error ? error.message : error}</p>
+      <p>{error instanceof Error ? error.message : String(error)}</p>
```
[the remark this answers is not in the document]
Comment (stemey) [reply packages/demos/src/react-loading-and-error-ui/integrator.tsx:20]: yep

File: packages/demos/src/history-service/history-consumer/index.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import {FeatureAppDefinition, FeatureServices} from '@feature-hub/core';
 import {HistoryServiceV3} from '@feature-hub/history-service';
 import {ReactFeatureApp} from '@feature-hub/react';
 import * as React from 'react';
+
```
[the remark this answers is not in the document]
Comment (stemey) [reply packages/demos/src/history-service/history-consumer/index.tsx]: removed

**Review by clebert — APPROVED** [review]

**Pull request merged**